### DISCUSSION TITLE
Sentiment analysis for all standard spacy language models

### DISCUSSION THREAD

**Discussion by thomasdhuw** [explosion/spaCy#12655] [Language Support]
Hi,

I was hoping to perform sentiment analysis against a variety of languages that spacy supports.

It works fine with english - I followed the example from https://spacy.io/universe/project/spacy-textblob

```
import spacy
from spacytextblob.spacytextblob import SpacyTextBlob

nlp = spacy.load('en_core_web_sm')
nlp.add_pipe('spacytextblob')
text = 'I had a really horrible day. It was the worst day ever! But every now and then I have a really good day that makes me happy.'
doc = nlp(text)
doc._.blob.polarity
doc._.blob.subjectivity
```

I was hoping that I could load a different language model and it would work out of the box, but it doesn't appear to. All scores come back as 0 for non-english models.

Is there a way I can have sentiment analysis on all the standard models (e.g. en_core_web_md and fi_core_news_sm and it_core_news_sm, etc)?

Thanks

**Comment by vinbo8** [comment]
spacytextblob is a third-party package, and they appear to support mainly only English, and French and German via [extensions](https://textblob.readthedocs.io/en/dev/extensions.html#extensions). SpaCy models are very task-dependent and do not come with sentiment analysis out-of-the-box, so I would suggest looking around for other third-party packages that can do this.
